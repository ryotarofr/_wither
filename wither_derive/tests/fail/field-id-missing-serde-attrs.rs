use serde::{Serialize, Deserialize};
use wither::Model;

#[derive(Serialize, Deserialize, Model)]
struct BadModel {
    id: Option<wither::bson::oid::ObjectId>,
}

fn main() {}
