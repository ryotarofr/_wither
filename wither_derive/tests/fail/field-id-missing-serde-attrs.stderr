error: the ID field of wither::Models must have the attribute `#[serde(rename="_id", skip_serializing_if="Option::is_none")]`
 --> $DIR/field-id-missing-serde-attrs.rs:6:5
  |
6 |     id: Option<wither::bson::oid::ObjectId>,
  |     ^^
